gettsim/benefits/benefit_checks.py

```diff
@@ -1,153 +1,214 @@
 import numpy as np
+import pandas as pd
 
+from gettsim.dynamic_function_generation import create_function
 
-def benefit_priority(household, params):
-    """There are three main transfers for working-age people:
-        1. Basic social security / ALG2
-        2. Housing Benefit / Wohngeld
-        3. Additional Child Benefit / Kinderzuschlag
 
-    There is a rule which benefits are superior to others
-    If there is a positive ALG2 entitlement, but the need can be covered with
-    Housing Benefit (and possibly add. child benefit),
-    the HH *has* to claim the housing benefit and addit. child benefit.
-    If the household need cannot be covered via Wohngeld, he has to apply for ALG2.
-    There is no way you can receive ALG2 and Wohngeld/Kinderzuschlag at the same time!
+def freibetrag_alter(kind, alter, geburtsjahr, arbeitsl_geld_2_params):
+    """Calculate exemptions based on individuals age."""
+    conditions = [
+        geburtsjahr < 1948,
+        (1948 <= geburtsjahr) & ~kind,
+        True,
+    ]
+
+    choices = [
+        arbeitsl_geld_2_params["vermögensfreibetrag"]["vor_1948"] * alter,
+        arbeitsl_geld_2_params["vermögensfreibetrag"]["standard"] * alter,
+        0,
+    ]
+
+    data = np.select(conditions, choices)
+
+    return pd.Series(index=alter.index, data=data)
+
+
+def freibetrag_alter_per_hh(hh_id, freibetrag_alter):
```

Comment:
I realized that you named all variables `per_hh` or `per_tu` which belong on the household or tax_unit level. Lets agree to one layout. I suggest  just a simple `_tu` or `_hh` at the end of the variable.